src/snowflake/snowpark/modin/plugin/extensions/indexing_overrides.py

```diff
@@ -1192,9 +1192,10 @@ def __getitem__(
         if not isinstance(col_loc, pd.Series) and is_range_like(col_loc):
             col_loc = self._convert_range_to_valid_slice(col_loc)
 
-        # Convert all scalar, list-like, and indexer row_loc to a Series object to get a query compiler object.
+        # Convert scalar to slice to generate efficient SQL query
         if is_scalar(row_loc):
-            row_loc = pd.Series([row_loc])
+            row_loc = slice(row_loc, None if row_loc == -1 else row_loc + 1, 1)
```

Comment:
ideally, this could be just ==, instead of a filter, right? do we use == for any case?

Reply:
yes, we don't have `==` used anywhere today so this can be the simplest way to improve.